src/main/java/dev/slimevr/vr/processor/skeleton/SimpleSkeleton.java

```diff
@@ -407,17 +411,29 @@ protected void updateLocalTransforms() {
 			headNode.localTransform.setRotation(Quaternion.IDENTITY);
 		}
 		
-		if(chestTracker.getRotation(rotBuf1)) {
-			neckNode.localTransform.setRotation(rotBuf1);
+		// Spine
+		if(hasSpineTracker){
+			if(chestTracker.getRotation(rotBuf1)) {
+				neckNode.localTransform.setRotation(rotBuf1);
+			}
+			if(waistTracker.getRotation(rotBuf1)) {
+				chestNode.localTransform.setRotation(rotBuf1);
+				trackerChestNode.localTransform.setRotation(rotBuf1);
+			}
+			if(hipTracker.getRotation(rotBuf1)) {
+				waistNode.localTransform.setRotation(rotBuf1);
+				hipNode.localTransform.setRotation(rotBuf1);
+				trackerWaistNode.localTransform.setRotation(rotBuf1);
+			}
 		}
-		if(waistTracker.getRotation(rotBuf1)) {
+		else if(hmdTracker != null){ // If no spine tracker, allign spine yaw with HMD
+			rotBuf1 = new Quaternion().fromAngles(0, FastMath.atan2(rotBuf1.getY(), rotBuf1.getW())*2f, 0);
```

Comment:
This is completely wrong way to get a yaw...

Reply:
Why? atan2(rotBuf1.getY(), rotBuf1.getW()) works...
How would you do it?

Reply:
There is a method `Quaternion.toAngles(float[3])` that sets provided array to [pitch, yaw, roll]. `atan2(rotBuf1.getY(), rotBuf1.getW())` doesn't cover all cases. Also, you're recreating rotBuf1, instead of just doing `rotBuf1.fromAngels(...)`.

Reply:
Would you prefer `rotBuf1 = rotBuf1.fromAngles(0, rotBuf1.toAngles(null)[1], 0);` ?
I can do that but it just seems more expensive to compute each tick...

Reply:
> There is a method `Quaternion.toAngles(float[3])` that sets provided array to [pitch, yaw, roll]. `atan2(rotBuf1.getY(), rotBuf1.getW())` doesn't cover all cases. Also, you're recreating rotBuf1, instead of just doing `rotBuf1.fromAngels(...)`.

I wrote the above message before seeing this btw. I'll do that 1 sec

Reply:
I would prefer `rotBuf1.fromAngles(0, rotBuf1.toAngles(anglesBuf)[1], 0);` and declare `angelsBuf` as a private static final to re-use. It's more expensive to calculate all angles, but is more correct. Even more correct would be to add a getYaw() method to Quaternion: https://github.com/Eirenliel/slime-java-commons/commit/94acd0dfaee0dca5ad51811c8f3bcf8facc964f2 :P

Reply:
Alright, used GetYaw(). Should be good now (tested in SteamVR too)

On another note, should I open another PR in the fixYaw method to use GetYaw now that we have that?

It does `		targetTrackerRotation.toAngles(angles);
		targetTrackerRotation.fromAngles(0, angles[1], 0);`

Reply:
You can, but the calibration should use another method via vectors anyway (this too, but the math framework haven't been figured out yet)